//[validator-core](../../../index.md)/[com.chrynan.validator](../index.md)/[ValidationException](index.md)



# ValidationException  
 [common] class [ValidationException](index.md) : [RuntimeException](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-runtime-exception/index.html)

A [RuntimeException](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-runtime-exception/index.html) that indicates there was an error asserting that a [ValidationResult](../-validation-result/index.md) was valid or invalid.



This exception is thrown from the [assertValid](../assert-valid.md) and [assertInvalid](../assert-invalid.md) functions.

   


## Properties  
  
|  Name |  Summary | 
|---|---|
| <a name="com.chrynan.validator/ValidationException/cause/#/PointingToDeclaration/"></a>[cause](index.md#%5Bcom.chrynan.validator%2FValidationException%2Fcause%2F%23%2FPointingToDeclaration%2F%5D%2FProperties%2F277157140)| <a name="com.chrynan.validator/ValidationException/cause/#/PointingToDeclaration/"></a> [common] open val [cause](index.md#%5Bcom.chrynan.validator%2FValidationException%2Fcause%2F%23%2FPointingToDeclaration%2F%5D%2FProperties%2F277157140): [Throwable](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html)?   <br>|
| <a name="com.chrynan.validator/ValidationException/message/#/PointingToDeclaration/"></a>[message](index.md#%5Bcom.chrynan.validator%2FValidationException%2Fmessage%2F%23%2FPointingToDeclaration%2F%5D%2FProperties%2F277157140)| <a name="com.chrynan.validator/ValidationException/message/#/PointingToDeclaration/"></a> [common] open val [message](index.md#%5Bcom.chrynan.validator%2FValidationException%2Fmessage%2F%23%2FPointingToDeclaration%2F%5D%2FProperties%2F277157140): [String](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html)?   <br>|

